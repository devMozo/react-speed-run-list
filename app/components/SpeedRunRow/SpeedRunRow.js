import PropTypes from 'prop-types';
import Link from 'next/link';
import ViewURL from 'app/constants/urls';
import './SpeedRunRow.scss';

const SpeedRunRow = ({
 style, data: {
 assets, names, released, id,
},
}) => (
  <li className="SpeedRunRow" style={style}>
    <picture className="SpeedRunRow__logo">
      <img src={assets['cover-medium'].uri} alt={names.international} />
    </picture>
    <section className="SpeedRunRow__description">
      <h2>{names.international}</h2>
      <h4>
        Released:
        {'  '}
        {released}
      </h4>
    </section>
    <Link className="SpeedRunRow__link" href={`${ViewURL}/${id}`} prefetch>
      <button type="button">Go!</button>
    </Link>
  </li>
);

SpeedRunRow.propTypes = {
  style: PropTypes.object,
  data: PropTypes.object,
};

export default SpeedRunRow;
