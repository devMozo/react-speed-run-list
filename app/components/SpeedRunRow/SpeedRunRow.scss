@import '../../fonts/index.scss';

.SpeedRunRow {
  overflow: hidden;
  border-bottom: 1px solid #eee;
  box-sizing: border-box;
  font-family: $fontRegular;

  & > * {
    display: inline-block;
    vertical-align: middle;
  }

  .SpeedRunRow__description {
    padding: 0px 20px;
    box-sizing: border-box;
    & > * {
      width: 250px;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      margin: 0;
    }

    h2 {
      color: #333;
      font-size: 18px;
    }

    h4 {
      color: #888;
      font-size: 14px;
    }
  }

  .SpeedRunRow__logo {
    width: 100px;
    height: inherit;

    img {
      object-fit: cover;
      overflow: hidden;
      width: inherit;
      height: inherit;
    }
  }

  .SpeedRunRow__link {
    float: right;
    border: 0;
    background-color: #333;
    color: white;
    margin: 20px 30px 0 0px;
    padding: 10px 20px;
    cursor: pointer;
    transition: 0.3s;

    &:hover {
      opacity: 0.6;
    }
  }
}
