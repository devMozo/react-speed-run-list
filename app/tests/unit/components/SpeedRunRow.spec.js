import React from 'react';
import { shallow } from 'enzyme';
import SpeedRunRow from '../../../components/SpeedRunRow/SpeedRunRow';
import ViewURL from '../../../constants/urls';

describe('<SpeedRunRow />', () => {
  const MOCKED_PROPS = () => ({
    data: {
      assets: {
        'cover-medium': {
          uri: 'mocked-uri',
        },
      },
      names: {
        international: 'Mocked International Name',
      },
      released: 'Mocked Released',
      id: 123,
    },
  });

  it('should render correctly', () => {
    const component = shallow(<SpeedRunRow {...MOCKED_PROPS()} />);

    expect(component).toMatchSnapshot();
  });

  it('it must contain the correct fields', () => {
    const props = MOCKED_PROPS();
    const { data } = props;
    const component = shallow(<SpeedRunRow {...props} />);
    const img = component.find('.SpeedRunRow__logo img');
    const name = component.find('.SpeedRunRow__description h2');
    const released = component.find('.SpeedRunRow__description h4');
    const link = component.find('.SpeedRunRow__link');

    expect(img.prop('src')).toEqual(data.assets['cover-medium'].uri);
    expect(img.prop('alt')).toEqual(data.names.international);
    expect(name.text()).toEqual(data.names.international);
    expect(released.text()).toContain(data.released);
    expect(link.prop('href')).toEqual(`${ViewURL}/${data.id}`);
  });
});
