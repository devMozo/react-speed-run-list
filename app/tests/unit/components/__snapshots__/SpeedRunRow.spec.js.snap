// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpeedRunRow /> should render correctly 1`] = `
<li
  className="SpeedRunRow"
>
  <picture
    className="SpeedRunRow__logo"
  >
    <img
      alt="Mocked International Name"
      src="mocked-uri"
    />
  </picture>
  <section
    className="SpeedRunRow__description"
  >
    <h2>
      Mocked International Name
    </h2>
    <h4>
      Released:
        
      Mocked Released
    </h4>
  </section>
  <Link
    className="SpeedRunRow__link"
    href="/View/123"
    prefetch={true}
  >
    <button
      type="button"
    >
      Go!
    </button>
  </Link>
</li>
`;
